e2e/test/scenarios/embedding/embedding-dashboard.cy.spec.js

```diff
@@ -623,19 +623,29 @@ describe("scenarios > embedding > dashboard parameters with defaults", () => {
   });
 
   it("locked parameters require a value to be specified in the JWT", () => {
-    openStaticEmbeddingModal({ activeTab: "parameters" });
+    const nameParameter = dashboardDetails.parameters.find(
+      parameter => parameter.name === "Name",
+    );
+    const sourceParameter = dashboardDetails.parameters.find(
+      parameter => parameter.name === "Source",
+    );
 
-    // ID param is disabled by default
-    setEmbeddingParameter("Name", "Editable");
-    setEmbeddingParameter("Source", "Locked");
-    publishChanges("dashboard", ({ request }) => {
-      assert.deepEqual(request.body.embedding_params, {
-        source: "locked",
-        name: "enabled",
+    cy.get("@dashboardId").then(dashboardId => {
```

Comment:
Opt for setting up tests with API to speed up tests.

And modify the test behavior so that we go into the embed with `null` as a locked parameter value. We already do this for the native question test, so nothing has changed there.